src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -55,7 +56,11 @@
 
   private MethodSpec(Builder builder) {
     CodeBlock code = builder.code.build();
-    checkArgument(code.isEmpty() || !builder.modifiers.contains(Modifier.ABSTRACT),
+    checkArgument(!(builder.modifiers.contains(Modifier.ABSTRACT)
+        && hasDefaultModifier(builder.modifiers)),
+        "abstract method %s cannot be default", builder.name);
+    checkArgument(code.isEmpty() || !builder.modifiers.contains(Modifier.ABSTRACT)
+        || hasDefaultModifier(builder.modifiers),
```

Comment:
No, see checkArg above. Couldn't get the whole thing in a boolean statement, if someone has better logic than me feel free to offer up one.